### PR TITLE
add create_statefulset activity

### PR DESCRIPTION
**Pull request by tdevilleduc**
Signed-off-by: Thomas DEVILLE-DUC <thomas.deville-duc-prestataire@laposte.fr>

### CODE REVIEW

**Comment by Lawouach** [comment]
and the last one, with the requested changes (chanlelog and tests please) :)

**Comment by tdevilleduc** [comment]
> and the last one, with the requested changes (chanlelog and tests please) :)

For now, i don't know how to test it :)
But i will try something later

**Comment by Lawouach** [comment]
Okay, I'll wait for these tests:

* one that loads a json and asserts (via a mock) that the create_service was applied on the Kubernetes client (see for instance https://github.com/chaostoolkit/chaostoolkit-kubernetes/blob/master/tests/test_pod.py#L14)
* one that does the same from a yaml (store the json and yaml files in the tests/fixtures directory)
* one that triggers the ActivityFailed because it couldn't load the requested file
* maybe others where the yaml/jso nfiles are malformed so fail to be loaded too?

Also the changelog entry

File: chaosk8s/statefulset/actions.py [hunk 1]
```diff
@@ -1,4 +1,7 @@
 # -*- coding: utf-8 -*-
+import json
+import os.path
+import yaml
```
Comment (Lawouach) [inline chaosk8s/statefulset/actions.py]: that import should be after the kubernetes one as it's a third-party

**Comment by tdevilleduc** [comment]
> LGTM
> 
> Could you just move the files under the `fixtures` directory instead of creating the `file_mocks` one?
> 
> and also the changelog entry please :)

You're too fast :)
Should I just put files directly into 'fixtures' or should I keep statefulset/create folders ?

> that import should be after the kubernetes one as it's a third-party

Sorry. I didn't understand :(

File: chaosk8s/statefulset/actions.py (hunk 1 above)
Comment (Lawouach) [reply chaosk8s/statefulset/actions.py]: I should clarify, the `yaml` import is a third-party so its place should be after the last third-party imports in the next block.

We tend to group imports as follows:

stdlib

third-party

local

:)

**Comment by Lawouach** [comment]
Happy to keep the `statefulset/create` directory :)

File: chaosk8s/statefulset/actions.py (hunk 1 above)
Comment (tdevilleduc) [reply chaosk8s/statefulset/actions.py]: From other spots, third-party imports are above kubernetes (with a blank line separation)
https://github.com/chaostoolkit/chaostoolkit-kubernetes/blob/master/chaosk8s/pod/actions.py

Comment (Lawouach) [reply chaosk8s/statefulset/actions.py]: Yeap, they are wrong and should be fixed. Let's ensure yours is good from the get go :)

Comment (tdevilleduc) [reply chaosk8s/statefulset/actions.py]: Ok, so I can be sure 
Like this ?

> from chaoslib.exceptions import ActivityFailed
> from chaoslib.types import Secrets
> from kubernetes import client
> from kubernetes.client.rest import ApiException
> from logzero import logger
> 
> import json
> import os.path
> import yaml

Comment (tdevilleduc) [reply chaosk8s/statefulset/actions.py]: I searched a bit more for example and i change for this : 

> import json
> import os.path
> 
> from chaoslib.exceptions import ActivityFailed
> from chaoslib.types import Secrets
> from kubernetes import client
> from kubernetes.client.rest import ApiException
> from logzero import logger
> import yaml

json and os.path are stdlib, right ?
(Sorry I'm newbie in python)

Comment (Lawouach) [reply chaosk8s/statefulset/actions.py]: Yes. That's the right one. 

Indeed, json and os.path are stdlib :)

**Comment by Lawouach** [comment]
well done and merci for being patient with me all along!

**Pull request merged**